NLP/main/extract_display_quotes.py

```diff
@@ -77,164 +80,87 @@ def __init__(self):
+
+        self.file_uploader = widgets.FileUpload(
+            description='Upload your files (txt, csv or xlsx)',
+            accept='.txt, .xlsx, .csv ', # accepted file extension 
             multiple=True,  # True to accept multiple files
-            layout = widgets.Layout(width='280px')
-            )
-        
-        # widget to upload .xlsx file
-        uploader_xls = widgets.FileUpload(
-            description='Click here to upload .xlsx, .xls, or .csv file', 
-            accept='.xlsx, .xls, .csv', # accepted file extension
-            multiple=False,  # to accept one Excel file only
             layout = widgets.Layout(width='320px')
-            )
-        
-        # tab widget to select which file types to upload
-        children = [uploader_text, uploader_xls]
-        tab = widgets.Tab(layout = widgets.Layout(width='450px'))
-        tab.children = children
-        tab.set_title(0, 'Upload text files')
-        tab.set_title(1, 'Upload excel file')
-        
-        
-        # give notification when file is uploaded
-        def _cb(change):
-            clear_output()
-            uploaded_success = widgets.HTML(
-            value='<b>File uploaded!</b>',
-            placeholder='',
-            description='',
-            layout=Layout(margin='0px 0px 0px 0px')
-            )
-            display(uploaded_success)
-            
-            # process the uploaded file
-            if tab.children[0].value!={}:
-                self.process_txt(tab.children[0])
-            else:
-                self.process_xls(tab.children[1])
-            
-        uploader_text.observe(_cb, names='data')
-        uploader_xls.observe(_cb, names='data')
-        
-        return tab
+            ) 
+
     
+
     
-    def nlp_preprocess(self, text):
+    def load_txt(self, value):
+        # Load individual txt file content and return a dict object, wrapped in a list so it can be merged with list of pervious file contents.
+        temp = {'text_name': value['metadata']['name'][:-4],
+                'text': codecs.decode(value['content'], encoding='utf-8')
+        }
+        return [temp]
+
+
+    def load_table(self, value, file_fmt):
+        # Load csv or xlsx file
+        if file_fmt == 'csv':
+            temp_df = pd.read_csv(io.BytesIO(value['content']))
+        if file_fmt == 'xlsx':
+            temp_df = pd.read_excel(io.BytesIO(value['content']))
+        # Check if the column text and text_name present in the table, if not, skip the current spreadsheet
+        if ('text' not in temp_df.columns) or ('text_name' not in temp_df.columns):
+            print('File {} does not contain the required header "text" and "text_name"'.format(value['metadata']['name']))
+            return []
+        # Return a list of dict objects
+        temp = temp_df[['text_name', 'text']].to_dict(orient='index').values()
+        return temp
+
+
+    def hash_gen(self, temp_df):
```

Comment:
text_id doesn't seem to be working properly, duplicate id's when uploading many docs at the same time. also, text_id is just a number?
[quotes.xlsx](https://github.com/Sydney-Informatics-Hub/ATAP-GenderGapTracker/files/9069105/quotes.xlsx)

Reply:
I haven't touched the text_id part in the quote_df, but did notice it was just an interger number. I think the deduplication was working, but will double check. The text_id in quote_df should be looked in a different branch after merging, if the uploading function works as expected in this branch.

Reply:
Fixed the issue. It is related to not using text_id as index of the dataframe.